### ISSUE TITLE
Unable to access role history correctly

### ISSUE THREAD

**Issue by sbenthall** [glasgow-ipl/ietfdata#129]
I am trying to use this package to get records about working group leadership in the present and past.

Getting them for the present seems to work fine. As an example I'm using `dnsop`.
https://datatracker.ietf.org/group/dnsop/history/

```
> wg = dt.group_from_acronym("dnsop")
> [dt.person(r.person).name for r in dt.group_roles(group = wg)]
['Warren "Ace" Kumari', 'Benno Overeinder', 'Suzanne Woolf', 'Tim Wicinski']
```
These are indeed the names of the people currently in leadership of this group.

Now I'm trying to find the people historically but not currently in roles in this working group.

```
> [dt.person(r.person).name for r in dt.group_role_histories(group = wg)]
['Andrew Sullivan', 'Marc Blanchet']
```

I don't think these people have ever been in `dnsop` leadership. I'm expecting the names Peter Koch, Joel Jaeggli, and Ronald Bonica. These names are mentioned in certain GroupEvents about leadership change. Is there link to these other persons maintained in the group history?

**Comment by lumisota** [comment]
The `group` parameter for `group_role_histories` should be a `GroupHistoryURI`, and not a `GroupURI`. This isn't an intuitive name, but it does match the Datatracker API.

You can fetch the `GroupHistory` objects for a group like this:

```
wg = dt.group_from_acronym("dnsop")
group_role_histories = [dt.group_role_histories(group = grp_hist, name=dt.role_name(RoleNameURI("/api/v1/name/rolename/chair/"))) for grp_hist in group_histories]
```
(Note the use of `name` to filter to only WG chairs: you can remove this filter, or switch it to other roles (e.g., `ad`).)

There'll be a `GroupHistory` object for each time the metadata about a group changes, including for chair changes. This has evolved over time: the first entry I can see is from 2005.

From each `GroupHistory` object, you can look up the roles held at that point:

```
for group_role_history in group_role_histories:
    print([dt.person(r.person).name for r in group_role_history])
```

That should give you the WG chairs for each `GroupHistory`.

**Comment by sbenthall** [comment]
Thank you. That's so helpful!

Is there a way to get the `GroupHistoryURI` from a known `GroupURI`, or acronym?

**Comment by sbenthall** [comment]
Hmmm. I'm sorry, I think I asked a badly thought through question.

I am seeing how to get group histories from an acronym:
https://github.com/glasgow-ipl/ietfdata/blob/master/ietfdata/datatracker.py#L674

Which may be just what I want!

I'm a bit confused about what you are saying about `group_role_histories`.
Is it possible that the type signature of the method is off?
https://github.com/glasgow-ipl/ietfdata/blob/master/ietfdata/datatracker.py#L2883

It says that `group` is `Optional[Group'` not `Optional[GroupHistory]`

**Comment by lumisota** [comment]
Yes -- the signature is wrong here; should be `Optional[GroupHistory]` as you say. Will fix (along with the tests..).
